store/tikv/txn.go

```diff
@@ -469,7 +469,8 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *kv.LockCtx, keysInput .
 					txn.us.UnmarkPresumeKeyNotExists(key)
 				}
 			}
-			keyMayBeLocked := terror.ErrorNotEqual(kv.ErrWriteConflict, err) && terror.ErrorNotEqual(kv.ErrKeyExists, err)
+			_, isErrKeyExist := errors.Cause(err).(*ErrKeyExist)
+			keyMayBeLocked := terror.ErrorNotEqual(kv.ErrWriteConflict, err) && !isErrKeyExist
```

Comment:
Why do we need to use different ways here? The deadlock error below also uses `dl, ok := errors.Cause(err).(*ErrDeadlock);`

Reply:
Do you mean embedding the downcast in the `if` expression like
```
if _, isErrKeyExist := errors.Cause(err).(*ErrKeyExist); len(keys) > 1 || (terror.ErrorNotEqual(kv.ErrWriteConflict, err) && !isErrKeyExist) {
```

I'm afraid it looks too long...

Reply:
#23686 fixed the same bug on master with a helper function. So I change this PR to be consistent with the master.